app/src/main/java/org/schabi/newpipe/database/playlist/dao/PlaylistStreamDAO.java

```diff
@@ -87,21 +89,36 @@ default Flowable<List<PlaylistStreamEntity>> listByService(final int serviceId)
     Flowable<List<PlaylistStreamEntry>> getOrderedStreamsOf(long playlistId);
 
     @Transaction
-    @Query("SELECT " + PLAYLIST_ID + ", " + PLAYLIST_NAME + ", " + PLAYLIST_THUMBNAIL_URL + ", "
-            + "COALESCE(COUNT(" + JOIN_PLAYLIST_ID + "), 0) AS " + PLAYLIST_STREAM_COUNT
+    @Query("SELECT " + PLAYLIST_ID + ", " + PLAYLIST_NAME + ","
+
+            + " CASE WHEN " + PLAYLIST_THUMBNAIL_STREAM_ID + " = -1"
+            + " THEN " + "'" + DEFAULT_THUMBNAIL + "'"
+            + " ELSE (SELECT " + STREAM_THUMBNAIL_URL
+            + " FROM " + STREAM_TABLE
+            + " WHERE " + STREAM_TABLE + "." + STREAM_ID + " = " + PLAYLIST_THUMBNAIL_STREAM_ID
+            + " ) END AS " + PLAYLIST_THUMBNAIL_URL + ", "
 
+            + PLAYLIST_NAME + ", "
+            + "COALESCE(COUNT(" + JOIN_PLAYLIST_ID + "), 0) AS " + PLAYLIST_STREAM_COUNT
             + " FROM " + PLAYLIST_TABLE
             + " LEFT JOIN " + PLAYLIST_STREAM_JOIN_TABLE
-            + " ON " + PLAYLIST_ID + " = " + JOIN_PLAYLIST_ID
+            + " ON " + PLAYLIST_TABLE + "." + PLAYLIST_ID + " = " + JOIN_PLAYLIST_ID
             + " GROUP BY " + PLAYLIST_ID
             + " ORDER BY " + PLAYLIST_NAME + " COLLATE NOCASE ASC")
     Flowable<List<PlaylistMetadataEntry>> getPlaylistMetadata();
 
     @Transaction
     @Query("SELECT " + PLAYLIST_TABLE + "." + PLAYLIST_ID + ", "
             + PLAYLIST_NAME + ", "
-            + PLAYLIST_TABLE + "." + PLAYLIST_THUMBNAIL_URL + ", "
-            + "COALESCE(COUNT(" + JOIN_PLAYLIST_ID + "), 0) AS " + PLAYLIST_STREAM_COUNT + ", "
+
+            + " CASE WHEN " + PLAYLIST_THUMBNAIL_STREAM_ID + " = -1"
+            + " THEN " + "'" + DEFAULT_THUMBNAIL + "'"
+            + " ELSE (SELECT " + STREAM_THUMBNAIL_URL
+            + " FROM " + STREAM_TABLE
+            + " WHERE " + STREAM_TABLE + "." + STREAM_ID + " = " + PLAYLIST_THUMBNAIL_STREAM_ID
+            + " ) END AS " + PLAYLIST_THUMBNAIL_URL
+
+            + ", COALESCE(COUNT(" + JOIN_PLAYLIST_ID + "), 0) AS " + PLAYLIST_STREAM_COUNT + ", "
```

Comment:
Just for consistency
```suggestion
            + " ) END AS " + PLAYLIST_THUMBNAIL_URL + ", "

            + "COALESCE(COUNT(" + JOIN_PLAYLIST_ID + "), 0) AS " + PLAYLIST_STREAM_COUNT + ", "
```